src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -194,18 +199,22 @@ def parse(self, backend, x509_obj):
             except KeyError:
                 if critical:
                     raise x509.UnsupportedExtension(
-                        "{0} is not currently supported".format(oid), oid
+                        "Critical extension {0} is not currently supported"
+                        .format(oid), oid
                     )
             else:
-                d2i = backend._lib.X509V3_EXT_d2i(ext)
-                if d2i == backend._ffi.NULL:
-                    backend._consume_errors()
-                    raise ValueError(
-                        "The {0} extension is invalid and can't be "
-                        "parsed".format(oid)
-                    )
-
-                value = handler(backend, d2i)
+                if self.unsupported_exts and oid in self.unsupported_exts:
```

Comment:
Let's put a comment here to the effect of "for extensions that openssl doesn't directly support we pass the extension object itself through to the handler"